compiler/rustc_metadata/src/rmeta/decoder.rs

```diff
@@ -928,6 +928,14 @@ impl<'a, 'tcx> CrateMetadataRef<'a> {
             .map_id(|index| self.local_def_id(index))
     }
 
+    fn get_impl_restriction(self, id: DefIndex) -> ty::Restriction {
+        self.root
+            .tables
+            .impl_restriction
+            .get(self, id)
+            .map_or(ty::Restriction::Unrestricted, |restriction| restriction.decode(self))
```

Comment:
`.decode((self, sess))` like `get_expn_that_defined`, but you'll need to pass the session in the caller. 

try ` => { table }` first tho.